src/onyx/messaging/aeron.clj

```diff
@@ -181,25 +212,28 @@
 (defmethod extensions/send-messages AeronConnection
   [messenger event peer-link batch]
   (let [[len unsafe-buffer] (protocol/build-messages-msg-buf batch)
-        pub ^uk.co.real_logic.aeron.Publication (:pub peer-link)
+        pub ^uk.co.real_logic.aeron.Publication (:send-pub peer-link)
         offer-f (fn [] (.offer pub unsafe-buffer 0 len))]
     (while (not (offer-f)))))
 
 (defmethod extensions/internal-ack-message AeronConnection
   [messenger event peer-link message-id completion-id ack-val]
   (let [unsafe-buffer (protocol/build-acker-message message-id completion-id ack-val)
-        pub ^uk.co.real_logic.aeron.Publication (:pub peer-link)
+        pub ^uk.co.real_logic.aeron.Publication (:acker-pub peer-link)
         offer-f (fn [] (.offer pub unsafe-buffer 0 protocol/ack-msg-length))]
     (while (not (offer-f)))))
```

Comment:
This might be a good place to make use of the new to-be-implemented `retry` function on an input, rather than hammering away at backpressure.